modules/core/src/passes/pick-layers-pass.ts

```diff
@@ -150,6 +143,18 @@ export default class PickLayersPass extends LayersPass {
 
     return pickParameters;
   }
+
+  protected _resetColorEncoder(pickZ: boolean) {
+    // Track encoded layer indices
+    this._colors = pickZ
```

Comment:
Having a member called _colors that contains a complicated substructure instead of a colors array doesn't help. Since it is private, can we rename it something more descriptive?

Given that the method is called _resetColorEncoder, should we at least call the variable `_colorEncoder` or `_colorEncoderState` perhaps as it is not really an "encoder" by itself?